springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/scanners/bindings/BindingFactory.java

```diff
@@ -10,11 +10,11 @@
 import java.util.Map;
 
 public interface BindingFactory<T> {
-    default String getChannelId(T annotation) {
-        return ReferenceUtil.toValidId(getChannelName(annotation));
+    default String getChannelId(T annotation, Class<?> component) {
+        return ReferenceUtil.toValidId(getChannelName(annotation, component));
     }
 
-    String getChannelName(T annotation);
+    String getChannelName(T annotation, Class<?> component);
```

Comment:
This interface might be implemented by users, changing the interface breaks the contract.

Is it possible to create a new method instead and call the original method (delegate). Additionally, the original method can be commented with `maintainer note: replaced by ...` as done in other places, so that it will be cleanup in the next big release

Reply:
I wonder, whether this is actually similar to a context, like `MethodAndAnnotation`, but instead `ClassAndAnnotation` - while the annotation might be on the method OR class

Reply:
Looking at usages, using Method instead of Class might be better.
The class can always be found from a method AND a method provides a better context